### ISSUE TITLE
Confusing @import @extend %ruleset behavior

### ISSUE THREAD

**Issue by srcspider** [sass/sass#869]
_Some basic context: I came across this while trying to split the rules applied from the selectors by writing rules in a separate assets folder under %selectors and the actual selectors in an equity folder that mirrors the site page structure._

Sass 3.3.0.alpha.149 (Bleeding Edge)
Compass 0.12.2.sourcemaps (Alnilam)

To get the new source maps to work I'm combining my files in a single file by importing the relevant pages; and hence my problem.

The issue is that @import'ing %selector multiple times will duplicate the rules even when asking for compressed output. (it's clearer in the example)

Here is a basic example that reproduces the error:

```
/sass
  /rules
     test.scss
  x.scss
  y.scss
  all.scss
```

**test.scss**

``` css
%test {
    color: red;
}
```

**x.scss**

``` css
@import 'rules/test';

.x {
    @extend %test;
}
```

**y.scss**

``` css
@import 'rules/test';

.y {
    @extend %test;
}
```

**all.scss**

``` css
@import 'x';
@import 'y';
```

You will get the following faulty _all.css_

``` css
/* line 1, ../sass/rules/test.scss */
.x, .y {
  color: red;
}

/* line 1, ../sass/rules/test.scss */
.x, .y {
  color: red;
}
```

Expected: single `.x, .y` rule and if I have a rule in a file then no matter what @import structure I use it's still a single rule (since it doesn't make sense to have multiple copies). ie. the following `(%test, test.scss, color, red)` is unique.

**Comment by Snugug** [comment]
This is a duplicate of many other issues all requesting an import once feature. The issue where this is being tracked is #139

**Comment by srcspider** [comment]
@Snugug I can see how #139 _might_ fix it, but I don't see how the semantics of %selector in the context of an import are completely equivalent to an @import that doesn't run twice in all contexts, so I think this should stay open until #139 is comes to a conclusion/implementation.

For what it's worth even a `@depends` directive is implemented, why should the behavior I describe not work in the context of an `@import` when %select is meant to be a virtual selector; meaning it makes no sense for the same color definition in one instance to be considered different than the same color definition in another instance that was loaded in.

ie. even if I have TWO files with TWO `%something` if both have `color: red` there should be only one shared `color: red`.

**Comment by Snugug** [comment]
In which case, this becomes a potential duplicate of #116 as well.

I think you may be misunderstanding what `@extend` does.  It takes every instance of the item you're extending and comma separates it, allowing you to move selectors around. Think of the `%placeholder` selectors not as entirely different and novel concepts, bur rather as a class that simply won't print if it's not extended. In addition, Sass, in an effort not to not to misinterpret your meaning and potentially mess up your cascade, does exactly what you say. You said you wanted to import that placeholder selector twice, so it's going to. Removing one of the extends could negatively affect your cascade, so it currently doesn't do that. 

Moreover, the way you would solve your current issue would be as follows:

**all.scss**

``` scss
@import 'rules/test';
@import 'x';
@import 'y';
```

If this doesn't work for you because you are creating a framework, then you are precisely looking for the solution in #139. You would then write something like the following (syntax not final):

**x.scss**

``` scss
@import 'rules/test' !once;
.x {
    @extend %test;
}
```

**y.scss**

``` scss
@import 'rules/test' !once;
.y {
    @extend %test;
}
```

**Comment by srcspider** [comment]
(yes that's how I already solved it)

The problem with the solution is that it can only be ever used at the top level safely so it's more of a stop-gap measure (albeit I don't really have a problem with it since it's cleaner than having `@import`s in every file); anywhere else and you're likely to have someone `@import` it twice. The second `!once` solution has a similar semantic problem, if after the first `@import !once` someone then does `@import`'s **y.scss** twice the same undesirable effect would happen.

Should I convert the issue to a feature request for a alternative `@extend` or `%placeholder` syntax that ignores the cascade when generating? ie. "ignore the current position of the element; inject the element and all its children relative to the position of the %placeholder." Or in laymans terms the placeholder's position is the significant one, not the position of the element that calls `@extend`.

**Comment by nex3** [comment]
As @Snugug has mentioned, this is the expected behavior. If you import the file that defines `%placeholder` twice, then it will be defined twice and extended twice. In the future we'll move to an import system that will import a file only once by default.
